### PR TITLE
[REVIEW] Fix warnings 2

### LINKED ISSUE #958: Fix warnings in c++ code compilation
We ideally want to get to no WARNINGS during compilation. 
1. First lets start fixing all the real issues we have in our code
2. Identify what issues are caused by reasons outside our code that we cant feasably fix. 
3. Explore ways to have the compiler be silent about  the Warnings we can do nothing about

### PR DESCRIPTION
**Pull request by diegodfrf**
Issue #958

Add -Wall -Wextra to compile engine and io
800 Warnings - 33 warnings remainig
Add basic .clang-tidy file

### CODE REVIEW

**Comment by diegodfrf** [comment]
rerun-tests

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by diegodfrf** [comment]
rerun-tests

**Comment by rommelDB** [comment]
add to allowlist

**Comment by rommelDB** [comment]
ok to test

**Comment by aucahuasi** [comment]
add to allowlist

**Comment by diegodfrf** [comment]
rerun-tests

File: engine/src/CalciteInterpreter.cpp [hunk 1]
```diff
@@ -9,7 +9,7 @@ std::shared_ptr<ral::cache::graph> generate_graph(std::vector<ral::io::data_load
 	std::vector<std::string> table_names,
 	std::vector<std::string> table_scans,
 	std::string logicalPlan,
-	int64_t connection,
+	int64_t  /*connection*/,
```
Comment (williamBlazing) [inline engine/src/CalciteInterpreter.cpp:12]: if we are not using this parameter, we should just take it out

File: engine/src/cython/initialize.cpp [hunk 2]
```diff
@@ -522,9 +522,9 @@ ucp_ep_h CreateUcpEp(ucp_worker_h ucp_worker,
 * and the cache we use for receiving messages
 *
 */
-std::pair<std::pair<std::shared_ptr<CacheMachine>,std::shared_ptr<CacheMachine> >, int> initialize(int ralId,
+std::pair<std::pair<std::shared_ptr<CacheMachine>,std::shared_ptr<CacheMachine> >, int> initialize(uint16_t ralId,
 	std::string worker_id,
-	int gpuId,
+	int /*gpuId*/,
```
Comment (williamBlazing) [inline engine/src/cython/initialize.cpp:527]: if we are not using this parameter, which should just take it out

File: engine/src/execution_graph/logic_controllers/BatchJoinProcessing.cpp [hunk 3]
```diff
@@ -843,7 +843,7 @@ void JoinPartitionKernel::small_table_scatter_distribution(std::unique_ptr<ral::
 	std::string small_output_cache_name = scatter_left_right.first ? "output_a" : "output_b";
 	int small_table_idx = scatter_left_right.first ? LEFT_TABLE_IDX : RIGHT_TABLE_IDX;
 	std::string big_output_cache_name = scatter_left_right.first ? "output_b" : "output_a";
-	int big_table_idx = scatter_left_right.first ? RIGHT_TABLE_IDX : LEFT_TABLE_IDX;
+	//int big_table_idx = scatter_left_right.first ? RIGHT_TABLE_IDX : LEFT_TABLE_IDX;
```
Comment (williamBlazing) [inline engine/src/execution_graph/logic_controllers/BatchJoinProcessing.cpp:846]: if we are not using it, just delete it

File: engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.cpp [hunk 4]
```diff
@@ -343,14 +343,14 @@ kstatus PartitionKernel::run() {
     std::map<std::string, std::map<int32_t, int> > node_count;
 
     std::tie(sortColIndices, sortOrderTypes, std::ignore) =	ral::operators::get_sort_vars(this->expression);
-    auto& self_node = ral::communication::CommunicationData::getInstance().getSelfNode();
+    //auto& self_node = ral::communication::CommunicationData::getInstance().getSelfNode();
```
Comment (williamBlazing) [inline engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.cpp:346]: delete this

File: engine/src/io/DataLoader.cpp [hunk 5]
```diff
@@ -32,11 +32,11 @@ data_loader::~data_loader() {}
 
 
 std::unique_ptr<ral::frame::BlazingTable> data_loader::load_batch(
-	const std::vector<int> & column_indices_in,
-	const Schema & schema,
-	data_handle file_data_handle,
-	size_t file_index,
-	std::vector<cudf::size_type> row_group_ids) {
+	const std::vector<int> & /*column_indices_in*/,
```
Comment (williamBlazing) [inline engine/src/io/DataLoader.cpp:35]: I think this whole class is dead code. Please delete it

Comment (williamBlazing) [reply engine/src/io/DataLoader.cpp:35]: make sure it is first before deleting :-)

**Review by williamBlazing — CHANGES_REQUESTED** [review]

File: engine/src/CalciteInterpreter.cpp (hunk 1 above)
Comment (rommelDB) [reply engine/src/CalciteInterpreter.cpp]: Removed.

File: engine/src/cython/initialize.cpp (hunk 2 above)
Comment (rommelDB) [reply engine/src/cython/initialize.cpp]: Removed.

File: engine/src/execution_graph/logic_controllers/BatchJoinProcessing.cpp (hunk 3 above)
Comment (rommelDB) [reply engine/src/execution_graph/logic_controllers/BatchJoinProcessing.cpp]: Removed.

File: engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.cpp (hunk 4 above)
Comment (rommelDB) [reply engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.cpp]: Removed.

File: engine/src/io/DataLoader.cpp (hunk 5 above)
Comment (rommelDB) [reply engine/src/io/DataLoader.cpp]: It seems that only this function is deprecated. Maybe is still a valid abstraction for grouping parsers and providers.

**Comment by rommelDB** [comment]
rerun tests

**Review by felipeblazing — CHANGES_REQUESTED** [review]

File: engine/src/execution_graph/logic_controllers/LogicalProject.cpp [hunk 6]
```diff
@@ -523,6 +523,96 @@ std::unique_ptr<cudf::column> evaluate_string_functions(const cudf::table_view &
         );
         break;
     }
+    case operator_type::BLZ_INVALID_OP:
```
[the remark this answers is not in the document]
Comment (rommelDB) [reply engine/src/execution_graph/logic_controllers/LogicalProject.cpp]: Replaced.

File: engine/src/execution_graph/logic_controllers/CacheMachine.h [hunk 7]
```diff
@@ -796,8 +796,8 @@ class WaitingQueue {
 				bool done_waiting = this->finished.load(std::memory_order_seq_cst);
 				if (!done_waiting) {
 					size_t total_bytes = 0;
-					for (int i = 0; i < message_queue_.size(); i++){
-						total_bytes += message_queue_[i]->get_data().sizeInBytes();
+					for (auto & i : message_queue_){
+						total_bytes += i->get_data().sizeInBytes();
```
[the remark this answers is not in the document]
Comment (rommelDB) [reply engine/src/execution_graph/logic_controllers/CacheMachine.h]: Renamed.

File: engine/src/execution_graph/logic_controllers/CacheMachine.h [hunk 8]
```diff
@@ -796,8 +796,8 @@ class WaitingQueue {
 				bool done_waiting = this->finished.load(std::memory_order_seq_cst);
 				if (!done_waiting) {
 					size_t total_bytes = 0;
-					for (int i = 0; i < message_queue_.size(); i++){
-						total_bytes += message_queue_[i]->get_data().sizeInBytes();
+					for (auto & i : message_queue_){
```
[the remark this answers is not in the document]
Comment (rommelDB) [reply engine/src/execution_graph/logic_controllers/CacheMachine.h]: Renamed.

File: engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.cpp [hunk 9]
```diff
@@ -577,10 +576,10 @@ kstatus LimitKernel::run() {
 
         std::vector<std::string> limit_messages_to_wait_for;
         std::vector<std::string> target_ids;
-        for (auto i = 0; i < nodes_to_send.size(); i++) {
-            target_ids.push_back(nodes_to_send[i].id());
+        for (auto node_to_send : nodes_to_send) {
```
[the remark this answers is not in the document]
Comment (rommelDB) [reply engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.cpp]: Using reference.

File: engine/src/execution_graph/logic_controllers/BatchJoinProcessing.cpp [hunk 10]
```diff
@@ -647,10 +647,10 @@ std::pair<bool, bool> JoinPartitionKernel::determine_if_we_are_scattering_a_smal
 
 	std::vector<std::string> determination_messages_to_wait_for;
 	std::vector<std::string> target_ids;
-	for (auto i = 0; i < nodes_to_send.size(); i++)	{
-		target_ids.push_back(nodes_to_send[i].id());
+	for (auto node_to_send : nodes_to_send) {
```
[the remark this answers is not in the document]
Comment (rommelDB) [reply engine/src/execution_graph/logic_controllers/BatchJoinProcessing.cpp]: Using reference.

File: engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.cpp [hunk 11]
```diff
@@ -577,10 +576,10 @@ kstatus LimitKernel::run() {
 
         std::vector<std::string> limit_messages_to_wait_for;
         std::vector<std::string> target_ids;
-        for (auto i = 0; i < nodes_to_send.size(); i++) {
```
[the remark this answers is not in the document]
Comment (rommelDB) [reply engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.cpp:580]: Using reference.

File: engine/src/execution_graph/logic_controllers/BatchJoinProcessing.cpp [hunk 12]
```diff
@@ -647,10 +647,10 @@ std::pair<bool, bool> JoinPartitionKernel::determine_if_we_are_scattering_a_smal
...
+		target_ids.push_back(node_to_send.id());
```
[the remark this answers is not in the document]
Comment (rommelDB) [reply engine/src/execution_graph/logic_controllers/BatchJoinProcessing.cpp:651]: Using reference.

**Comment by rommelDB** [comment]
rerun tests

**Pull request merged**